rxgroups-android/src/main/java/com/airbnb/rxgroups/android/GroupLifecycleManager.java

```diff
@@ -0,0 +1,172 @@
+
+      if (hashCode != observableManager.hashCode()) {
+        group = observableManager.newGroup();
+      } else {
+        ObservableGroup group = observableManager.getGroup(groupId);
+        if (group.isDestroyed()) {
+          Log.w(TAG, "Tried to reuse GroupLifecycleManager with a destroyed group");
+          group = observableManager.newGroup();
+          shouldResubscribe = false;
+        }
+        this.group = group;
+      }
+    }
+
+    if (shouldResubscribe) {
+      subscribe(target);
+    }
+  }
+
+  /**
+   * Returns whether the provided {@link Class} exists for the {@link ObservableGroup}.
+   * Observables will only be removed from their respective groups once {@link
+   * Observer#onCompleted()} has been called.
+   */
+  public boolean hasObservable(Class<?> klass) {
+    return hasObservable(tagFactory.tag(klass));
+  }
+
+  public boolean hasObservable(String tag) {
+    return group.hasObservable(tag);
+  }
+
+  /**
+   * Subscribe all Observer fields on the target that are annotated with {@link AutoResubscribe}
+   * and that have their corresponding Observable in flight.
+   */
+  public void subscribe(Object target) {
+    Subscription subscription = resubscription.observers(target)
+        .filter(new Func1<LifecycleResubscription.ObserverInfo, Boolean>() {
+          @Override public Boolean call(LifecycleResubscription.ObserverInfo observerInfo) {
+            return hasObservable(observerInfo.klass);
+          }
+        }).subscribe(new Action1<LifecycleResubscription.ObserverInfo>() {
+          @Override public void call(LifecycleResubscription.ObserverInfo observerInfo) {
+            resubscribe(observerInfo.klass, observerInfo.listener);
+          }
+        });
+
+    pendingSubscriptions.add(subscription);
+  }
+
+  /**
+   * Resubscribes an existing {@link Class} If the Observable has already emitted events,
+   * they will be immediately delivered if the group is
+   * unlocked. Any previously subscribed Observers will be unsubscribed before the new one.
+   */
+  public void resubscribe(Class<?> klass, Observer<?> observer) {
+    resubscribe(tagFactory.tag(klass), observer);
+  }
+
+  @SuppressWarnings({ "unchecked", "rawtypes" })
+  public void resubscribe(String tag, Observer<?> observer) {
+    Observable observable = group.observable(tag);
+    observable.subscribe(observer);
+  }
+
+  /**
+   * Cancel the given request if it is running, so that no future request callbacks are received.
+   * If the request is not running then nothing
+   * happens.
+   */
+  public void cancelAndRemove(Class<?> klass) {
+    group.cancelAndRemove(tagFactory.tag(klass));
+  }
+
+  private void onCreate() {
+    hasSavedState = false;
+  }
+
+  private void onDestroy(boolean isFinishing) {
+    pendingSubscriptions.clear();
+
+    if (isFinishing) {
+      observableManager.destroy(group);
+    } else {
+      group.unsubscribe();
+    }
+  }
+
+  public void onDestroy(@Nullable Activity activity) {
```

Comment:
We should add documentation about how the destroy calls work, why we track `hasSavedState`, and how we decide to  destroy the underlying observable manager.

We should maybe also address how fragment management isn't perfect - the main case I can think of where this falls apart is for fragments in a pager where the fragment is destroyed when it is off screen and we'll never probably destroy the group.